sierra_adapter/sierra_item_merger/src/test/scala/uk/ac/wellcome/platform/sierra_item_merger/dynamo/MergedSierraRecordDaoTest.scala

```diff
@@ -0,0 +1,191 @@
+import uk.ac.wellcome.models.aws.DynamoConfig
+import uk.ac.wellcome.platform.sierra_item_merger.locals.DynamoDBLocal
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.dynamo._
+
+class MergedSierraRecordDaoTest
+    extends FunSpec
+    with Matchers
+    with DynamoDBLocal
+    with ScalaFutures
+    with MockitoSugar
+    with ExtendedPatience {
+
+  val mergedSierraRecordDao =
+    new MergedSierraRecordDao(dynamoDbClient, DynamoConfig(tableName))
+
+  describe("get a merged sierra record") {
+    it(
+      "should return a future of merged sierra record if it exists in  DynamoDB") {
+      val mergedSierraRecord = MergedSierraRecord(id = "b1111",
+                                                  maybeBibData = None,
+                                                  itemData = Map(),
+                                                  version = 1)
+
+      Scanamo.put(dynamoDbClient)(tableName)(mergedSierraRecord)
+
+      whenReady(mergedSierraRecordDao.getRecord(mergedSierraRecord.id)) {
+        record =>
+          record shouldBe Some(mergedSierraRecord)
+      }
+    }
+
+    it(
+      "should return a future of None if the record does not exist in DynamoDB") {
+      val mergedSierraRecord = MergedSierraRecord(id = "b1111",
+                                                  maybeBibData = None,
+                                                  itemData = Map(),
+                                                  version = 1)
+
+      Scanamo.put(dynamoDbClient)(tableName)(mergedSierraRecord)
+
+      whenReady(mergedSierraRecordDao.getRecord("b88888")) { record =>
+        record shouldBe None
+      }
+    }
+
+    it(
+      "returns a failed future with the underlying exception if the DynamoDB read fails") {
+      val dynamoDbClient = mock[AmazonDynamoDB]
+      val expectedException = new RuntimeException("AAAAAARGH!")
+      when(dynamoDbClient.getItem(any[GetItemRequest]))
+        .thenThrow(expectedException)
+
+      val mergedSierraRecordDaoMockedDynamoClient =
+        new MergedSierraRecordDao(dynamoDbClient, DynamoConfig(tableName))
+
+      val future =
+        mergedSierraRecordDaoMockedDynamoClient.getRecord("b88888")
+
+      whenReady(future.failed) { ex =>
+        ex shouldBe expectedException
+      }
+    }
+  }
+
+  describe("update a merged sierra record") {
+    it(
+      "should insert a new merged sierra record if it doesn't already exist in DynamoDB") {
+      val id = "b1111"
+      val mergedSierraRecord = MergedSierraRecord(
+        id = id,
+        maybeBibData = None,
+        itemData = Map(
+          "i111" -> SierraItemRecord(id = "i111",
+                                     data = "something",
+                                     modifiedDate = "2001-01-01T01:01:30Z",
+                                     bibIds = List("b1111"))),
+        version = 0
+      )
+
+      whenReady(mergedSierraRecordDao.updateRecord(mergedSierraRecord)) { _ =>
+        Scanamo
```

Comment:
You may want to nick `dynamoQueryEqualsValue` from the bib merger feature tests, or put that into a common lib – would allow you to clean up this code somewhat.